ethereum/GhostOwnership.sol

```diff
@@ -0,0 +1,96 @@
+pragma solidity ^0.5.0;
+
+import "./GhostFactory.sol";
+import "./ERC721.sol";
+
+contract GhostOwnership is GhostFactory, ERC721 {
+    function _owns(address _owner, uint256 _tokenId) internal view returns (bool) {
+        return ghostIndexToOwner[_tokenId] == _owner;
+    }
+
+    function balanceOf(address _owner) public view returns (uint256 count) {
+        return ownershipTokenCount[_owner];
+    }
+
+    function _approvedFor(address _approved, uint256 _tokenId) internal view returns (bool) {
+        return ghostIndexToApproved[_tokenId] == _approved;
+    }
+
+    function _approve(address _approved, uint256 _tokenId) internal {
+        ghostIndexToApproved[_tokenId] = _approved;
+    }
+
+    function transfer(address _to, uint256 _tokenId) external {
+        require(_to != address(this));
+        require(_owns(msg.sender, _tokenId));
+
+        _transfer(msg.sender, _to, _tokenId);
+    }
+
+    function approve(address _to, uint256 _tokenId) external {
+        require(_owns(msg.sender, _tokenId));
+
+        _approve(_to, _tokenId);
+
+        Approval(msg.sender, _to, _tokenId);
+    }
+
+    function transferFrom(address _from, address _to, uint256 _tokenId) external {
+        require(_to != address(this));
+        require(_approvedFor(msg.sender, _tokenId));
+        require(_owns(_from, _tokenId));
+
+        _transfer(_from, _to, _tokenId);
+    }
+
+    function totalSupply() public view returns (uint) {
+        return ghosts.length - 1;
+    }
+
+    // TODO: calculate ratio of angel and devil
+    function getGauge() public view returns (uint, uint) {
+        return (0, 0);
+    }
+    // TODO: distinguish angel and devil
+    function distinguishAngelAndDevil() internal view returns (bool) {
```

Comment:
It is a function that distinguishes whether ghost is an angel or a devil.
I created the function because implementation might be complicated.
This function will be used in getGauge().